File: Backend/services/vneid.js
/**
 * VNeID Mock Service
 * This simulates VNeID API integration for demo purposes
 * In production, this would connect to the actual VNeID API
 */

/**
 * Verify user with VNeID
 * @param {string} cccdNumber - Citizen ID number
 * @returns {Promise<Object|null>} - VNeID user data or null if verification fails
 */
async function verifyVNeID(cccdNumber) {
  // Simulate API call delay
  await new Promise(resolve => setTimeout(resolve, 500));

  // Mock verification logic
  // In production, this would make an actual API call to VNeID service
  // For demo, we'll accept any valid 12-digit CCCD number
  
  if (!cccdNumber || cccdNumber.length !== 12 || !/^\d+$/.test(cccdNumber)) {
    return null;
  }

  // Mock VNeID response
  // In real implementation, this would be the actual response from VNeID API
  return {
    id: `vneid_${cccdNumber}`,
    cccdNumber: cccdNumber,
    verified: true,
    verifiedAt: new Date().toISOString(),
    // Additional VNeID data that might be returned
    fullName: null, // Would be populated from VNeID
    dateOfBirth: null,
    address: null
  };
}

/**
 * Get VNeID user information
 * @param {string} vneidId - VNeID user ID
 * @returns {Promise<Object|null>} - User information from VNeID
 */
async function getVNeIDUserInfo(vneidId) {
  // Simulate API call delay
  await new Promise(resolve => setTimeout(resolve, 300));

  if (!vneidId) {
    return null;
  }

  // Mock VNeID user info
  return {
    id: vneidId,
    verified: true,
    verificationLevel: 'level2', // level1, level2, level3
    verifiedAt: new Date().toISOString()
  };
}

/**
 * Check if VNeID is available (for demo purposes, always return true)
 * In production, this would check if VNeID service is online
 */
async function isVNeIDAvailable() {
  return true;
}

module.exports = {
  verifyVNeID,
  getVNeIDUserInfo,
  isVNeIDAvailable
};

File: Backend/services/vneid.py
"""
VNeID verification service.

TODO: Thay thế bằng VNeID API thật khi có credentials:
  - Endpoint: https://dichvucong.gov.vn/api/vneid/verify
  - Docs: https://dichvucong.gov.vn/developer

Hiện tại: mock cho phép demo/dev.
"""
import os
from datetime import datetime


def verify_vneid(cccd_number: str) -> dict | None:
    """Xác thực CCCD qua VNeID. Mock trả về thông tin demo."""
    if not cccd_number or len(str(cccd_number)) != 12 or not str(cccd_number).isdigit():
        return None
    return {
        'id':           f'vneid_{cccd_number}',
        'cccdNumber':   cccd_number,
        'verified':     True,
        'verifiedAt':   datetime.utcnow().isoformat() + 'Z',
        'fullName':     'Nguyễn Văn A',
        'dateOfBirth':  '1990-01-01',
        'address':      'Thanh Hóa, Việt Nam',
    }


def get_vneid_user_info(vneid_id: str) -> dict | None:
    if not vneid_id:
        return None
    return {
        'id':                vneid_id,
        'verified':          True,
        'verificationLevel': 'level2',
        'verifiedAt':        datetime.utcnow().isoformat() + 'Z',
    }


def is_vneid_available() -> bool:
    return True


# Backward-compat aliases (tên cũ camelCase)
verifyVNeID       = verify_vneid
getVNeIDUserInfo  = get_vneid_user_info
isVNeIDAvailable  = is_vneid_available
